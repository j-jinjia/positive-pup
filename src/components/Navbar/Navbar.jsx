import "./Navbar.scss";
import logo from "../../assets/images/navbar-logo.png";
import openMenu from "../../assets/svgs/hamburger-menu.svg";
import dropdownArrow from "../../assets/svgs/dropdown-arrow.svg";
import closeMenu from "../../assets/svgs/close-menu.svg";
import phone from "../../assets/svgs/phone.svg";
import { useState } from "react";
import Button from "../Button/Button";
import DropdownMenu from "../DropdownMenu/DropdownMenu";
import { Link, NavLink } from "react-router-dom";
import useWindowSize from "../../hooks/useWindowSize.js";
import { useEffect } from "react";

const Navbar = () => {
  const [showMenu, setShowMenu] = useState(false);
  const [showDropdown, setShowDropdown] = useState(false);
  const windowIsDesktop = useWindowSize(1024);

  useEffect(() => {
    if (showMenu && windowIsDesktop) {
      setShowMenu(false);
    }
  }, [windowIsDesktop]);
  const handleClick = () => {
    setShowMenu(!showMenu);
  };

  const toggleDropdown = () => {
    setShowDropdown(!showDropdown);
  };

  const desktopNav = (
    <nav className="navbar__nav">
      <NavLink className="navbar__nav-item" to="/about">
        About
      </NavLink>
      <div className="navbar__nav-item navbar__nav-item--dropdown">
        <button className="navbar__nav-item-button" onClick={toggleDropdown}>
          Services
          <img src={dropdownArrow} className="navbar__nav-item-arrow" />
        </button>
        {showDropdown && <DropdownMenu toggleDropdown={toggleDropdown} />}
      </div>
      <span className="navbar__nav-item navbar__nav-item--phone">
        <img className="navbar__nav-item-button" src={phone} alt="" />
        <NavLink className="navbar__nav-item" to="/get-in-touch">
          Get in Touch
        </NavLink>
      </span>
      <Button
        className="navbar__nav-item"
        link={<Link to={"/book-now"}>Book Now</Link>}
      />
    </nav>
  );

  const mobileNav = (
    <nav className="navbar__nav">
      <Link className="navbar__nav-item" to="/">
        Home
      </Link>
      <Link className="navbar__nav-item" to="/about">
        About
      </Link>
      <Link className="navbar__nav-item" to="/services">
        Services
      </Link>
      <Link className="navbar__nav-item" to="/get-in-touch">
        Get in Touch
      </Link>
      <Link
        className="navbar__nav-item navbar__nav-item--active"
        to="/book-now"
      >
        Book Now
      </Link>
    </nav>
  );

  const menuIcon = showMenu ? closeMenu : openMenu;
  const navbarClass = showMenu ? "navbar navbar--active" : "navbar";
  const navbarJSX = (
    <div className={navbarClass}>
      <div className="navbar__container">
        <Link to="/">
          <img
            src={logo}
            className="navbar__container-logo"
            alt="The Positive Pup logo"
          />
        </Link>

        {windowIsDesktop ? (
          desktopNav
        ) : (
          <button onClick={handleClick} className="navbar__container-button">
            <img
              src={menuIcon}
              className="navbar__container-button-image"
              alt=""
            />
          </button>
        )}
      </div>
      {!windowIsDesktop && showMenu && mobileNav}
    </div>
  );

  return navbarJSX;
};

export default Navbar;
