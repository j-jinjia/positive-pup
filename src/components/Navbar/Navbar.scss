@use "../../styles/_variables.scss" as *;

.navbar {
  width: 100%;
  padding: 1rem 2rem;
  background-color: white;

  &--active {
    height: 100vh;
    position: fixed;
    z-index: 1;
    top: 0;
    left: 0;
  }

  &__container {
    display: flex;
    justify-content: space-between;
    width: 100%;

    &-button {
      background: transparent;
      border: none;

      &:hover {
        cursor: pointer;
      }
    }
  }

  &__nav {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-end;
    height: 100%;

    &-item {
      &--active {
        font-weight: bold;
        color: $colour-primary;
      }
    }

    &-item,
    &-item:visited {
      padding: 21px 0;
      text-decoration: none;
      font-size: $font-size-navbar;
      border-bottom: 1px solid #f7f7f7;
      width: 100%;
      text-align: right;
    }
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .navbar {
    &__container {
      display: flex;
      flex-direction: row;
      justify-content: flex-start;
      align-items: center;
    }

    &__nav {
      flex-direction: row;
      justify-content: flex-start;
      align-items: center;
      width: 100%;
      gap: 50px;

      &-item {
        text-align: center;
        white-space: nowrap;
        font-size: $font-size-general;
        border: none;
        padding: 0;
        width: fit-content;
        &:nth-child(1) {
          margin-left: 50px;
        }
        &:nth-child(5) {
          margin-left: auto;
          display: flex;
          justify-content: center;
          align-items: center;
          gap: 12px;
        }
      }
    }
  }
}
