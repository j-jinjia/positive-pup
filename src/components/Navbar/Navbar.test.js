import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Navbar from "./Navbar";

it("should match the Navbar component snapshot", () => {
  const { container } = customRender(<Navbar />);

  expect(container).toMatchSnapshot();
});

it("services dropdown should be visible when button clicked", () => {
  customRender(<Navbar />);
  const button = screen.getByRole("button");
  userEvent.click(button);
  const viewAll = screen.getByText("View All");
  expect(viewAll).toBeInTheDocument();
});
