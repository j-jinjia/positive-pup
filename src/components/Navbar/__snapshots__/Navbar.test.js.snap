// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the Navbar component snapshot 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="navbar__container"
    >
      <a
        href="/"
      >
        <img
          alt="The Positive Pup logo"
          class="navbar__container-logo"
          src="navbar-logo.svg"
        />
      </a>
      <nav
        class="navbar__nav"
      >
        <a
          class="navbar__nav-item"
          href="/about"
        >
          About
        </a>
        <a
          class="navbar__nav-item"
          href="/services"
        >
          Services
        </a>
        <a
          class="navbar__nav-item"
          href="/testimonials"
        >
          Testimonials
        </a>
        <a
          class="navbar__nav-item"
          href="/gallery"
        >
          Gallery
        </a>
        <span
          class="navbar__nav-item navbar__nav-item--phone"
        >
          <img
            alt=""
            class="navbar__nav-item-button"
            src="phone.svg"
          />
          <a
            class="navbar__nav-item"
            href="/get-in-touch"
          >
            Get in Touch
          </a>
        </span>
        <div
          class="button button--primary button--link"
        >
          <a
            href="/book-now"
          >
            Book Now
          </a>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
